assets/app/data/crowbar/services/upgrade-status.factory.spec.js

```diff
@@ -187,6 +187,61 @@ describe('Upgrade Status Factory', function () {
                     });
                 });
             });
+
+            describe('when got upgrade status from api unsuccessfully', function () {
+                describe('when timeout == 0', function () {
+                    beforeEach(function () {
+                        bard.mockService(upgradeFactory, {
+                            getStatus: $q.reject()
+                        });
+
+                        upgradeStatusFactory.waitForStepToEnd(
+                            testedStep, mockedSuccessCallback, mockedErrorCallback, pollingInterval
+                        );
+
+                        $rootScope.$digest();
+                    });
+
+                    it('should not call success callback', function () {
+                        expect(mockedSuccessCallback).not.toHaveBeenCalled();
+                    });
+
+                    it('should call error callback', function () {
+                        expect(mockedErrorCallback).toHaveBeenCalled();
```

Comment:
Please, validate it has been called once. Also, it would be great to also assert the right `errorResponse` is being sent as a parameter
https://github.com/Itxaka/crowbar-ui/blob/c6682481862e7d0c2bb2e4ff3ce9b3837b2ab25a/assets/app/data/crowbar/services/upgrade-status.factory.js#L49